js/apps/system/_admin/aardvark/APP/react/src/views/query/slowQueries/SlowQueryHistory.tsx

```diff
@@ -0,0 +1,167 @@
+import {
+  Button,
+  ModalFooter,
+  ModalHeader,
+  Table,
+  TableCaption,
+  TableContainer,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr
+} from "@chakra-ui/react";
+import { QueryInfo } from "arangojs/database";
+import React from "react";
+import moment from "../../../../../frontend/js/lib/moment.min";
+import { Modal } from "../../../components/modal";
+import { getCurrentDB } from "../../../utils/arangoClient";
+import { useFetchSlowQueries } from "./useFetchSlowQueries";
+
+const TABLE_COLUMNS = [
+  {
+    Header: "ID",
+    accessor: "id"
+  },
+  {
+    Header: "Query String",
+    accessor: "query"
+  },
+  {
+    Header: "Bind Parameters",
+    accessor: "bindVars",
+    accessorFn: (row: any) => {
+      return JSON.stringify(row.bindVars);
+    }
+  },
+  {
+    Header: "User",
+    accessor: "user"
+  },
+  {
+    Header: "Peak Memory Usage",
+    accessor: "peakMemoryUsage"
+  },
+  {
+    Header: "Runtime",
+    accessor: "runTime",
+    accessorFn: (row: any) => {
+      return row.runTime.toFixed(2) + "s";
+    }
+  },
+  {
+    Header: "Started",
+    accessor: "started",
+    accessorFn: (row: any) => {
+      return moment(row.started).format("YYYY-MM-DD HH:mm:ss");
+    }
+  }
+];
+export const SlowQueryHistory = () => {
+  const { runningQueries, refetchQueries } = useFetchSlowQueries();
+  const [showDeleteModal, setShowDeleteModal] = React.useState(false);
+  return (
+    <>
+      <TableContainer backgroundColor="white">
+        <Table>
+          <Thead>
+            {TABLE_COLUMNS.map(column => {
```

Comment:
`TABLE_COLUMNS.map`, `runningQueries.map` and `runningQueries.length === 0` case are very similar to the cases found at `RunningQueries.tsx` file. Maybe it is possible to find an abstraction for them.

Reply:
Good catch! We have a ReactTable abstraction which wasn't available in devel when we started writing this code. Moved it to use that now, in this commit: [`22d4223` (#19380)](https://github.com/arangodb/arangodb/pull/19380/commits/22d4223c24336e04c22607304c662a9b94cf99f4)